core/model/src/main/java/org/eclipse/sw360/antenna/model/util/ArtifactLicenseUtils.java

```diff
@@ -54,13 +57,33 @@ public static LicenseInformation getFinalLicenses(Artifact artifact) {
         final Optional<LicenseInformation> declared = artifact.askForGet(DeclaredLicenseInformation.class);
         final Optional<LicenseInformation> observed = artifact.askForGet(ObservedLicenseInformation.class);
         if(declared.isPresent() && observed.isPresent()) {
-            final LicenseStatement effective = new LicenseStatement();
-            effective.setLeftStatement(declared.get());
-            effective.setRightStatement(observed.get());
-            effective.setOp(LicenseOperator.AND);
-            return effective;
-        }
+            List<License> allLicenses = new ArrayList<>(
+                    Stream.of(declared.get().getLicenses(), observed.get().getLicenses())
+                            .flatMap(List::stream)
+                            .collect(Collectors.toSet()));
 
+            return createLicenseStatement(allLicenses);
+        }
         return declared.orElse(observed.orElse(new LicenseStatement()));
     }
+
+    private static LicenseInformation createLicenseStatement(List<License> allLicenses) {
```

Comment:
```suggestion
    private static LicenseInformation createLicenseStatement(ArrayList<License> allLicenses) {
```
otherwise this will fail once an unmutable list is passed.

Essentially this is just a reduce, e.g.
```
   .reduce(
       LicenseStatement(),
       (lsmt1, lsmt2) -> {
           licenseStatement.setLeftStatement(lsmt1);
￼           licenseStatement.setOp(LicenseOperator.AND);
￼           licenseStatement.setRightStatement(lsmt2);
       });
```